### ISSUE TITLE
Crash with 'None' has no attribute 'meta'

### ISSUE THREAD

**Issue by rhdtownsend** [cmacmackin/ford#111]
In running FORD on the attached collection of source files (from my GYRE project; see https://bitbucket.org/rhdtownsend/gyre/wiki/Home), I get the following error:

```
Processing documentation comments...
Correlating information from different parts of your project...

Warning: Could not substitute link [[linalg]]. "linalg" not found.
Creating HTML documentation...
Traceback (most recent call last):
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/ford/output.py", line 89, in __init__
    self.docs.append(TypePage(data,project,item))
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/ford/output.py", line 184, in __init__
    self.html = self.render(data,proj,obj)
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/ford/output.py", line 305, in render
    return template.render(data,dtype=obj,project=proj)
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/jinja2/environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/jinja2/environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/ford/templates/type_page.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/ford/templates/base.html", line 99, in top-level template code
    {% block body %}
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/ford/templates/type_page.html", line 44, in block "body"
    {{ macros.interface(dtype.constructor) }}
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/ford/templates/macros.html", line 476, in template
    {{ proc_summary(proc.procedure) }}
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/ford/templates/macros.html", line 378, in template
    {% if small %}<h5>{% else %}<h3>{% endif %}{{ proc_line(proc,proto=proto) }}{% if proto %}<small>Prototype</small>{% endif %}{% if small %}</h5>{% else %}</h3>{% endif %}
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/ford/templates/macros.html", line 511, in template
    {% if proc.meta['deprecated'] and proc.meta['deprecated'].lower() == 'true' %}
  File "/Users/townsend/anaconda/lib/python3.5/site-packages/jinja2/environment.py", line 389, in getitem
    return obj[argument]
jinja2.exceptions.UndefinedError: 'None' has no attribute 'meta'
Error encountered.
```

cheers,

Rich

[gyre.zip](https://github.com/cmacmackin/ford/files/138473/gyre.zip)

**Comment by cmacmackin** [comment]
I've tracked down this error as arising form the constructor for the `cheby_t` type (in the file `gyre_cheby.f90`. The interface refers to `cheby_t_func_`, but this function is commented out. That fact is what causes FORD to crash. How on Earth can a compiler actually handle this file?

**Comment by rhdtownsend** [comment]
Ah — apologies, I commented this procedure out to work around the other problem with the function definition, which Izaak and I have been discussing.

cheers,

Rich

**Comment by cmacmackin** [comment]
Okay, now that I've fixed #110 and uncommented that function, everything works fine. I'll mentally file this under the category of "make more informative error messages."
